tests/queries/0_stateless/02434_cancel_insert_when_client_dies.sh

```diff
@@ -0,0 +1,98 @@
+$CLICKHOUSE_CLIENT -q "create table dedup_dist(A Int64) Engine = Distributed('test_cluster_one_shard_two_replicas', currentDatabase(), dedup_test)"
+
+function insert_data
+{
+    SETTINGS="query_id=$ID&max_insert_block_size=110000&min_insert_block_size_rows=110000"
+    # max_block_size=10000, so external table will contain smaller blocks that will be squashed on insert-select (more chances to catch a bug on query cancellation)
+    TRASH_SETTINGS="query_id=$ID&input_format_parallel_parsing=0&max_threads=1&max_insert_threads=1&max_insert_block_size=110000&max_block_size=10000&min_insert_block_size_bytes=0&min_insert_block_size_rows=110000&max_insert_block_size=110000"
+    TYPE=$(( RANDOM % 5 ))
+
+    if [[ "$TYPE" -eq 0 ]]; then
+        # client will send 10000-rows blocks, server will squash them into 110000-rows blocks (more chances to catch a bug on query cancellation)
+        $CLICKHOUSE_CLIENT --max_block_size=10000 --max_insert_block_size=10000 --query_id="$ID" \
+            -q 'insert into dedup_test settings max_insert_block_size=110000, min_insert_block_size_rows=110000 format TSV' < $DATA_FILE
+    elif [[ "$TYPE" -eq 1 ]]; then
+        $CLICKHOUSE_CLIENT --max_block_size=10000 --max_insert_block_size=10000 --query_id="$ID" --prefer_localhost_replica="$(( RANDOM % 2))" \
+            -q 'insert into dedup_dist settings max_insert_block_size=110000, min_insert_block_size_rows=110000 format TSV' < $DATA_FILE
+    elif [[ "$TYPE" -eq 2 ]]; then
+        $CLICKHOUSE_CURL -sS -X POST --data-binary @- "$CLICKHOUSE_URL&$SETTINGS&query=insert+into+dedup_test+format+TSV" < $DATA_FILE
+    elif [[ "$TYPE" -eq 3 ]]; then
+        $CLICKHOUSE_CURL -sS -X POST -H "Transfer-Encoding: chunked" --data-binary @- "$CLICKHOUSE_URL&$SETTINGS&query=insert+into+dedup_test+format+TSV" < $DATA_FILE
+    else
+        $CLICKHOUSE_CURL -sS -F 'file=@-' "$CLICKHOUSE_URL&$TRASH_SETTINGS&file_format=TSV&file_types=UInt64" -X POST --form-string 'query=insert into dedup_test select * from file' < $DATA_FILE
+    fi
+}
+
+export -f insert_data
+
+ID="02434_insert_init_${CLICKHOUSE_DATABASE}_$RANDOM"
+insert_data
+$CLICKHOUSE_CLIENT -q "system flush distributed dedup_dist"
+$CLICKHOUSE_CLIENT -q 'select count() from dedup_test'
+
+function thread_insert
+{
+    # supress "Killed" messages from bash
+    while true; do
+        export ID="$TEST_MARK$RANDOM"
+        bash -c insert_data 2>&1| grep -Fav "Killed"
+    done
+}
+
+function thread_select
+{
+    while true; do
+        $CLICKHOUSE_CLIENT -q "with (select count() from dedup_test) as c select throwIf(c != 5000000, 'Expected 5000000 rows, got ' || toString(c)) format Null"
+        sleep 0.$RANDOM;
+    done
+}
+
+function thread_cancel
+{
+    while true; do
+        SIGNAL="INT"
+        if (( RANDOM % 2 )); then
+            SIGNAL="KILL"
+        fi
+        PID=$(ps -ef | grep "$TEST_MARK" | grep -v grep | awk '{print $2}')
```

Comment:
Can we use `pidof` instead?

Reply:
Can we grep for `TEST_MARK` using `pidof`?

Reply:
`grep -l --text 'test' /proc/*/cmdline`